tests/v2/fixtures.py

```diff
@@ -0,0 +1,219 @@
+            except StopAsyncIteration:
+                break
+            response = from_client_interpreter(received=parsed, expected=next_expected)
+            if response is not None:
+                await server_to_client.put(response)
+            processing_finished.set()
+
+    server_task = asyncio.create_task(messages_from_client())
+
+    # server-to-client handler
+    #
+    # Consume ToClient events, optionally emitting datagrams to be written directly to
+    # the client.
+    #
+    # This represents the other half of the "server" lifecycle, where the server can
+    # choose to emit events directly without a request.
+
+    messages_to_client_channel = Channel[ToClient]()
+
+    async def messages_to_client() -> None:
+        our_task = asyncio.current_task()
+        while our_task and not our_task.cancelled():
+            try:
+                next_expected = await anext(messages_to_client_channel)
+            except StopAsyncIteration:
+                break
+
+            response = to_client_interpreter(next_expected)
+            if response is not None:
+                await server_to_client.put(response)
+            processing_finished.set()
+
+    processor_task = asyncio.create_task(messages_to_client())
+
+    # This consumes from the "expected" queue and routes messages to waiting channels
+    #
+    # This also handles shutdown
+
+    async def driver() -> None:
+        while expected:
+            next_expected = expected.popleft()
+            if isinstance(next_expected, FromClient):
+                await messages_from_client_channel.put(next_expected)
+            elif isinstance(next_expected, ToClient):
+                await messages_to_client_channel.put(next_expected)
+            elif isinstance(next_expected, WaitForClosed):
+                countdown = 100
+                messages_to_client_channel.close()
+                while not processor_task.done():
+                    if countdown <= 0:
+                        break
+                    countdown -= 1
+                    await asyncio.sleep(0.1)
+                client_to_server.close()
+                await client.close()
+                break
+            else:
+                assert_never(next_expected)
+            await asyncio.sleep(0.1)
+            await processing_finished.wait()
+
+    driver_task = asyncio.create_task(driver())
+
+    # Watchdog keeps track of the above tasks
+    # async def watchdog() -> None:
+    #     while True:
+    #         print(repr(server_task))
+    #         print(repr(processor_task))
+    #         print(repr(driver_task))
+    #         await asyncio.sleep(1)
+    #
+    # watchdog_task = asyncio.create_task(watchdog())
+
+    async def uri_and_metadata_factory() -> UriAndMetadata[None]:
+        return uri_and_metadata
+
+    client = Client(
+        uri_and_metadata_factory=uri_and_metadata_factory,
+        client_id="client-001",
+        server_id="server-001",
```

Comment:
Probably not super great to be hard-coded. Future versions of this test harness may expose some kind of Client Builder so that we can do request/response interleaving, should that be necessary to express.